common4j/src/main/com/microsoft/identity/common/java/net/IRetryPolicy.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.net;
+
+import java.io.IOException;
+import java.util.concurrent.Callable;
+
+/**
+ * Really a marker interface that takes a callable returning a type, and tries to produce
+ * an object of that type from it.  Implementations of this interface may examine the object
+ * returned and any exceptions that result from the call and decide to execute the supplier
+ * again in order to achieve a different result.
+ * @param <T> the type of the object on return.
+ */
+public interface IRetryPolicy<T> {
+    /**
+     * Evaluate the object returned from a callable and return the result.
+     * @param supplier an object to call for a result.
+     * @return the result of calling the supplier.
+     * @throws IOException if an IO error occurs.
+     */
+    T attempt (Callable<T> supplier) throws IOException;
```

Comment:
>T attempt (Callable<T> supplier) throws IOException; [](http://example.com/codeflow?start=4&length=52)

do we plan to use this for anything other than Http?
If so then maybe we need to move it out of the om.microsoft.identity.common.java.net pacakge

If not I think we can just create a HttpRetryPolicy Otherwise it looks to be over engineering for something that we might not use

Reply:
Same reason as below. Intentionally kept as-is to reduce the scope of the work.

We can refactor this later.